### PR TITLE
Added an extra volume to the azure VMs

### PR DESCRIPTION
**Pull request by yaso195**
By setting disk_size to a value more than 0, an additional disk
can be attached to a libretto azure VM.

@lilirui @zquestz

### CODE REVIEW

**Comment by lilirui** [comment]
Do we support attach multiple disks more than one?

**Comment by yaso195** [comment]
no, only one for now. There is not a easy way to write loops in arm templates. So, if we would like to support multiple disks, we may end up writing multiple templates for each different disk size.

File: virtualmachine/azure/arm/vm.go [hunk 1]
```diff
@@ -84,6 +84,10 @@ type VM struct {
 	// VM OS Properties
 	OsFile string
 
+	// VM Disk Properties
+	DiskFile string
```
Comment (lilirui) [inline virtualmachine/azure/arm/vm.go]: Nit: maybe you should comment the unit of the size

File: virtualmachine/azure/arm/util.go [hunk 2]
```diff
@@ -228,7 +241,14 @@ func (vm *VM) deleteOSFile(authorizer *azure.ServicePrincipalToken) error {
 
 	blobStorageClient := storageClient.GetBlobService()
 	err = blobStorageClient.DeleteBlob(vm.StorageContainer, vm.OsFile, nil)
-	return err
+	if err != nil {
+		return err
+	}
+	if vm.DiskSize == "" || vm.DiskSize == "0" {
```
Comment (lilirui) [inline virtualmachine/azure/arm/util.go]: This check is not very safe, can you convert string into int and then validate?

File: virtualmachine/azure/arm/vm.go (hunk 1 above)
Comment (yaso195) [inline virtualmachine/azure/arm/vm.go]: Done

File: virtualmachine/azure/arm/util.go (hunk 2 above)
Comment (yaso195) [inline virtualmachine/azure/arm/util.go]: Done

**Comment by yaso195** [comment]
Done, also updated the README

**Comment by lilirui** [comment]
LGTM

**Pull request merged**